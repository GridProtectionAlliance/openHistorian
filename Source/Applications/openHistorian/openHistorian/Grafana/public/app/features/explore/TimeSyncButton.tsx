import React from 'react';
import classNames from 'classnames';
import { css } from 'emotion';

import { Tooltip, useTheme, stylesFactory } from '@grafana/ui';
import { GrafanaTheme } from '@grafana/data';

const getStyles = stylesFactory((theme: GrafanaTheme) => {
  return {
    timePickerSynced: css`
      label: timePickerSynced;
      border-color: ${theme.colors.orangeDark};
      background-image: none;
      background-color: transparent;
      color: ${theme.colors.orangeDark};
      &:focus,
      :hover {
        color: ${theme.colors.orangeDark};
        background-image: none;
        background-color: transparent;
      }
    `,
    noRightBorderStyle: css`
      label: noRightBorderStyle;
      border-right: 0;
    `,
    /*
     * Required top-padding, otherwise is fa-link icon in active state
     * cut off on top due to fontAwesome icon position
     */
    topPadding: css`
      label: topPadding;
      padding-top: 1px;
    `,
  };
});

interface TimeSyncButtonProps {
  isSynced: boolean;
  onClick: () => void;
}

export function TimeSyncButton(props: TimeSyncButtonProps) {
  const { onClick, isSynced } = props;
  const theme = useTheme();
  const styles = getStyles(theme);

  const syncTimesTooltip = () => {
    const { isSynced } = props;
    const tooltip = isSynced ? 'Unsync all views' : 'Sync all views to this time range';
    return <>{tooltip}</>;
  };

  return (
    <Tooltip content={syncTimesTooltip} placement="bottom">
      <button
        className={classNames('btn navbar-button navbar-button--attached', {
          [`explore-active-button-glow ${styles.timePickerSynced}`]: isSynced,
        })}
        aria-label={isSynced ? 'Synced times' : 'Unsynced times'}
        onClick={() => onClick()}
      >
        <i className={classNames('fa fa-link', styles.topPadding, isSynced && 'icon-brand-gradient')} />
      </button>
    </Tooltip>
  );
}
