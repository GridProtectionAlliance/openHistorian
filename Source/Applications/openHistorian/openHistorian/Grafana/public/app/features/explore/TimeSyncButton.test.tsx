import React from 'react';
import { TimeSyncButton } from './TimeSyncButton';
import { mount } from 'enzyme';

const setup = (isSynced: boolean) => {
  const onClick = () => {};
  return mount(<TimeSyncButton onClick={onClick} isSynced={isSynced} />);
};

describe('TimeSyncButton', () => {
  it('should render component', () => {
    const wrapper = setup(true);
    expect(wrapper).toMatchSnapshot();
  });
  it('should change style when synced', () => {
    const wrapper = setup(true);
    expect(wrapper.find('button').props()['aria-label']).toEqual('Synced times');
  });
  it('should not change style when not synced', () => {
    const wrapper = setup(false);
    expect(wrapper.find('button').props()['aria-label']).toEqual('Unsynced times');
  });
});
