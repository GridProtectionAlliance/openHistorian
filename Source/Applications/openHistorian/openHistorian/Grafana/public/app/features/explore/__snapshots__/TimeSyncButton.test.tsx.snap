// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeSyncButton should render component 1`] = `
<TimeSyncButton
  isSynced={true}
  onClick={[Function]}
>
  <Component
    content={[Function]}
    placement="bottom"
  >
    <PopoverController
      content={[Function]}
      placement="bottom"
    >
      <button
        aria-label="Synced times"
        className="btn navbar-button navbar-button--attached explore-active-button-glow css-14r9fpj-timePickerSynced"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <i
          className="fa fa-link css-xmj56l-topPadding icon-brand-gradient"
        />
      </button>
    </PopoverController>
  </Component>
</TimeSyncButton>
`;
